src/pruna/evaluation/metrics/metric_vbench_background_consistency.py

```diff
@@ -0,0 +1,148 @@
+    # explicitly sets the device to cuda. We respect this here.
+    runs_on: List[str] = ["cuda"]
+    modality: List[str] = ["video"]
+    # state
+    similarity_scores: torch.Tensor
+    n_samples: torch.Tensor
+
+    def __init__(
+        self,
+        *args: Any,
+        device: str | None = None,
+        call_type: str = SINGLE,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+
+        if device is not None and str(device).split(":")[0] not in self.runs_on:
+            pruna_logger.error(f"Unsupported device {device}; supported: {self.runs_on}")
+            raise ValueError()
+
+        if call_type == PAIRWISE:
+            # VBench itself does not support pairwise.
+            # We can work on this in the future.
+            pruna_logger.error("VBench does not support pairwise metrics. Please use single mode.")
+            raise ValueError()
+
+        submodules_dict = init_submodules([METRIC_VBENCH_BACKGROUND_CONSISTENCY])
+        model_path = submodules_dict[METRIC_VBENCH_BACKGROUND_CONSISTENCY][0]
+
+        self.device = set_to_best_available_device(device)
+        self.call_type = get_call_type_for_single_metric(call_type, self.default_call_type)
+
+        self.clip_model, self.preprocessor = clip.load(model_path, device=self.device)
+        self.video_transform = clip_transform(224)
+
+        self.add_state("similarity_scores", torch.tensor(0.0))
+        self.add_state("n_samples", torch.tensor(0))
+
+    def update(self, x: List[str], gt: Any, outputs: Any) -> None:
+        """
+        Update the similarity scores for the batch.
+
+        Parameters
+        ----------
+        x : List[Any] | torch.Tensor
+            The input data.
+        gt : torch.Tensor
+            The ground truth / cached images.
+        outputs : torch.Tensor
+            The output images.
+        """
+        outputs = metric_data_processor(x, gt, outputs, self.call_type, device=self.device)
+        # Background consistency metric only supports a batch size of 1.
+        # To support larger batch sizes, we stack the outputs.
+        outputs = super().validate_batch(outputs[0])
+        # This metric depends on the outputs being uint8.
+        outputs = torch.stack([convert_image_dtype(output, dtype=torch.uint8) for output in outputs])
+        outputs = torch.stack([self.video_transform(output) for output in outputs])
+        features = torch.stack([self.clip_model.encode_image(output) for output in outputs])
+        features = torch.stack([F.normalize(feature, dim=-1, p=2) for feature in features])
+
+        # We vectorize the calculation to avoid for loops.
+        first_feature = features[:, 0, ...].unsqueeze(1).repeat(1, features.shape[1] - 1, 1)
+
+        similarity_to_first = F.cosine_similarity(first_feature, features[:, 1:, ...], dim=-1).clamp(min=0.0)
+        similarity_to_prev = F.cosine_similarity(features[:, :-1, ...], features[:, 1:, ...], dim=-1).clamp(min=0.0)
```

Comment:
How about nagative similarities. are we considering them to be zero? Should we use absolute values instead?